packages/dds/merge-tree/src/revertibles.ts

```diff
@@ -82,12 +95,41 @@ export interface MergeTreeRevertibleDriver {
  * exported for test only. should not be exported out the the package
  * @internal
  */
-export type InternalRevertDriver = MergeTreeRevertibleDriver & {
-	__mergeTreeRevertible?: {
-		detachedReferences?: EndOfTreeSegment;
-		refCallbacks?: LocalReferencePosition["callbacks"];
+export interface RevertRootMergeBlock {
+	mergeTree: MergeTree;
+	__mergeTreeRevertible: {
+		detachedReferences: EndOfTreeSegment;
+		refCallbacks: LocalReferencePosition["callbacks"];
 	};
-};
+}
+
+function findRevertRootMergeBlock(trackable: Trackable): RevertRootMergeBlock {
+	const segmentOrNode = trackable.isLeaf() ? trackable : trackable.getSegment();
+	const maybeRoot: Partial<RevertRootMergeBlock> | undefined = findRootMergeBlock(segmentOrNode);
+	assert(maybeRoot?.mergeTree !== undefined, "foo");
```

Comment:
nit: should the error message be more descriptive here?

Reply:
```suggestion
	assert(maybeRoot?.mergeTree !== undefined, "trackable is invalid as it is not in a rooted merge tree.");
```